### ISSUE TITLE
`@stylistic/unit-case` silently drops a fix in an at-rule whose name is not in lower case

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#104]
Found while fixing #99, and not a defect of that class: the rule picks its at-rules case-insensitively but writes the fix only to a node whose name is spelled in lower case, so an at-rule named in any other case is reported and then silently passed over.

`lib/rules/unit-case/index.js:118` selects with `/^media$/iu`, while the fixer at `lib/rules/unit-case/index.js:105` compares the name exactly:

```js
if (`name` in node && node.name === `media`) node.params = parsedValue.toString()
else if (`value` in node) node.value = parsedValue.toString()
```

An `@MEDIA` node therefore falls into the second branch, an ordinary at-rule has no `value` property of its own, and the assignment creates one that nothing prints. The write reaches no output while Stylelint counts the problem as fixed and drops the warning with it.

### What minimal example or steps are needed to reproduce the bug?

```css
@MEDIA (min-width: 10PX) {}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": "@stylistic/stylelint-plugin",
  "rules": {
    "@stylistic/unit-case": "lower"
  }
}
```

### What did you expect to happen?

`--fix` rewrites the unit — `@MEDIA (min-width: 10px) {}` — or, failing that, leaves the warning standing.

### What actually happened?

Without `--fix` the rule reports correctly, at `1:22`. With `--fix` the file is unchanged **and the warning is gone**, so a second run reports nothing to fix. The same query spelled `@media` is fixed as it should be.

### Do you have a proposal to fix the bug?

Have the fixer ask what the node is rather than what it is called — `isAtRule(node)` — and go through `syncLessVariableValue`, which the rule's second branch works around by hand for a Less at-variable. The whole branch then reads as one line for the params and one for the value.

Setting the params of a node the rule reached through the `variable` half of the same selection is harmless: `postcss-less` prints the value for such a node either way.
